detection_rules/schemas/v7_12.py

```diff
@@ -5,10 +5,55 @@
 
 """Definitions for rule metadata and schemas."""
 
+import jsl
+
+from .v7_9 import ThresholdMapping
 from .v7_11 import ApiSchema711
 
 
 class ApiSchema712(ApiSchema711):
     """Schema for siem rule in API format."""
 
     STACK_VERSION = "7.12"
+
+    # there might be a bug in jsl that requires us to redefine these here
+    query_scope = ApiSchema711.query_scope
+    saved_id_scope = ApiSchema711.saved_id_scope
+    ml_scope = ApiSchema711.ml_scope
+    eql_scope = ApiSchema711.eql_scope
+
+    class ThresholdMappingV12(ThresholdMapping):
+        """7.12 schema for threshold mapping."""
+
+        class ThresholdCardinality(jsl.Document):
+            """Threshold cardinality field."""
+
+            field = jsl.StringField(required=True)
+            value = jsl.IntField(minimum=1, required=True)
+
+        field = jsl.ArrayField(jsl.StringField(required=True, default=""))
+        cardinality = jsl.DocumentField(ThresholdCardinality, required=False)
+
+    threshold_scope = ApiSchema711.threshold_scope
+    threshold_scope.threshold = jsl.DocumentField(ThresholdMappingV12, required=True)
+
+    @classmethod
+    def downgrade(cls, target_cls, document, role=None):
+        """Remove 7.12 additions from the rule."""
+        # ignore when this method is inherited by subclasses
+        if cls in (ApiSchema712, ApiSchema712.versioned()) and 'threshold' in document:
+            threshold_field = document['threshold']['field'].copy()
+
+            # attempt to convert threshold field to a string
+            if len(threshold_field) > 1:
```

Comment:
```suggestion
            if len(threshold_field) > 1 or "cardinality" in document["threshold"]:
```

Reply:
Those are separate, because they could have multiple `threshold.field` values and _not_ define a cardinality. This is why I just do a generic pop of that whole field afterwards. But I need to add `None`!

Reply:
But if you have a cardinality, then it can't be backported.
That's more my point

Reply:
well the whole point of downgrade is to convert it to make it compatible. If you don't want to lose `cardinality` you should never downgrade.

Just like we did with `subtechnique`

Reply:
subtechnique just changes some metadata. does removing cardinality change the behavior? And in a little or big way

Reply:
ebb9a4bafa5bd6d36eb8a72f77fb5349362fa307